middleware/src/middleware.go

```diff
@@ -0,0 +1,105 @@
+package middleware
+
+import (
+	"time"
+	//"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/rpcclient"
+	//"github.com/btcsuite/btcutil"
+	"github.com/fiatjaf/lightningd-gjson-rpc"
+	//"github.com/tidwall/gjson"
+	"github.com/digitalbitbox/bitbox-base/middleware/util/logging"
+	"github.com/sirupsen/logrus"
+)
+
+type MiddlewareInfoStruct struct {
+	Blocks         int64   `json:"blocks"`
+	Difficulty     float64 `json:"difficulty"`
+	LightningAlias string  `json:"alias"`
+}
+
+type Middleware struct {
+	info                                               MiddlewareInfoStruct
+	events                                             chan *MiddlewareInfoStruct
+	bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort string
+	lightningRpcPath                                   string
+	log                                                *logrus.Entry
+}
+
+// NewMiddleware creates a new instance of the middleware
+func NewMiddleware(bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort, lightningRpcPath string) *Middleware {
+	middleware := &Middleware{
+		events:             make(chan *MiddlewareInfoStruct),
+		bitcoinRpcUser:     bitcoinRpcUser,
+		bitcoinRpcPassword: bitcoinRpcPassword,
+		bitcoinRpcPort:     bitcoinRpcPort,
+		lightningRpcPath:   lightningRpcPath,
+		log:                logging.Get().WithGroup("middleware"),
+	}
+	return middleware
+}
+
+func (middleware *Middleware) testBitcoinRPC(bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort string) MiddlewareInfoStruct {
+	connCfg := rpcclient.ConnConfig{
+		HTTPPostMode: true,
+		DisableTLS:   true,
+		Host:         "127.0.0.1:" + bitcoinRpcPort,
+		User:         bitcoinRpcUser,
+		Pass:         bitcoinRpcPassword,
+	}
+	client, err := rpcclient.New(&connCfg, nil)
+	if err != nil {
+		middleware.log.WithError(err).Error("Failed to create new bitcoind rpc client")
+	}
+	//client is shutdown/deconstructed again as soon as this function returns
+	defer client.Shutdown()
+
+	//Get current block count.
+	blockCount, err := client.GetBlockCount()
+	if err != nil {
+		middleware.log.WithError(err).Error("No blockcount received")
+		blockCount = 0
```

Comment:
Optional: It feels a bit odd to see an assignment of a return value in an `if err != nil {...}` block..

Perhaps it might be clearer to let the `blockCount` var remain with its nil value (`0`):
```
var (
	blockCount int
	err        error
)
blockCount, err = client.GetBlockCount()
if err != nil {
	middleware.log.WithError(err).Error("No blockcount received")
}
```